test/utilities/git.jl

```diff
@@ -1,11 +1,300 @@
+            cd(f)
+            run(`git init`)
+            run(`touch foo.txt`)
+            run(`touch bar.txt`)
+            output = read(`git status`, String)
+            @test occursin(untracked_str, output)
+
+            CompatHelper.git_add()
+            output = read(`git status`, String)
+            @test !occursin("untracked", output)
+            @test occursin(committed_str, output)
+        end
+    end
+
+    @testset "with flags" begin
+        mktempdir() do f
+            cd(f)
+            run(`git init`)
+            run(`touch foo.txt`)
+            run(`touch bar.txt`)
+            output = read(`git status`, String)
+            @test occursin(untracked_str, output)
+
+            CompatHelper.git_add(; flags="-A")
+            output = read(`git status`, String)
+            @test !occursin(untracked_str, output)
+            @test occursin(committed_str, output)
+        end
+    end
+
+    @testset "with items" begin
+        mktempdir() do f
+            cd(f)
+            run(`git init`)
+            run(`touch foo.txt`)
+            run(`touch bar.txt`)
+            run(`touch baz.txt`)
+            output = read(`git status`, String)
+            @test occursin(untracked_str, output)
+
+            CompatHelper.git_add(; items="foo.txt bar.txt")
+
+            output = read(`git status baz.txt`, String)
+            @test occursin(untracked_str, output)
+            @test !occursin(committed_str, output)
+
+            output = read(`git status foo.txt bar.txt`, String)
+            @test !occursin(untracked_str, output)
+            @test occursin(committed_str, output)
+        end
+    end
+end
+
+@testset "get_git_name_and_email" begin
+    @testset "using defaults" begin
+        expected = (
+            CompatHelper.COMPATHELPER_GIT_COMMITTER_NAME,
+            CompatHelper.COMPATHELPER_GIT_COMMITTER_EMAIL
         )
 
-        @test !isempty(readdir(local_path))
+        results = CompatHelper.get_git_name_and_email()
+        @test results == expected
+    end
+
+    @testset "setting env vars" begin
+        expected = ("User", "Email")
+
+        withenv(
+            "GIT_COMMITTER_NAME" => "User",
+            "GIT_COMMITTER_EMAIL" => "Email"
+        ) do
+            results = CompatHelper.get_git_name_and_email()
+            @test results == expected
+        end
+    end
+end
+
+@testset "git_clone" begin
+    @testset "HTTPS" begin
+        mktempdir() do f
+            cd(f)
+            local_path = joinpath(f, CompatHelper.LOCAL_REPO_NAME)
+            CompatHelper.git_clone(
+                "https://github.com/JuliaRegistries/CompatHelper.jl/", local_path
+            )
+
+            @test !isempty(readdir(local_path))
+        end
+    end
+
+    @testset "SSH" begin
+        mktempdir() do f
+            cd(f)
+            local_path = joinpath(f, CompatHelper.LOCAL_REPO_NAME)
+
+            CompatHelper.git_clone(
+                "git@github.com:JuliaRegistries/CompatHelper.jl.git",
```

Comment:
This fails because git needs the SSH key to be registered for even a clone of a repo. Not sure how to test this otherwise at this point, aside from just mocking the ```run(`git clone $url $local_path`)``` to return nothing.

Reply:
Right, cause it would get cloned to the credentials of that user.
I'd say just go forward with Mocking it for times sake.

a) We know this command works
b) Setting up credentials of some sort to properly test it seems like a lot more work for this one small test

Reply:
Fair enough. Makes sense to me